src/behaviors/Visibility/Visibility.js

```diff
@@ -363,7 +364,8 @@ export default class Visibility extends Component {
     const { bottom, height, top, width } = this.ref.getBoundingClientRect()
     const [topOffset, bottomOffset] = normalizeOffset(offset)
 
-    const direction = window.pageYOffset > this.pageYOffset ? 'down' : 'up'
+    const newOffset = (this.props.context ? this.props.context.scrollTop : window.pageYOffset)
```

Comment:
Since there are 4 instances of this, we should probably consider a method:

```jsx
getPageYOffset() {
  const { context } = this.props
  return context ? context.scrollTop : window.pageYOffset
}
```